readline/readline.go

```diff
@@ -60,20 +62,29 @@ func New(prompt Prompt) (*Instance, error) {
 }
 
 func (i *Instance) Readline() (string, error) {
+	if !i.Terminal.rawmode {
```

Comment:
would it be better to put these on methods of `Terminal`? e.g. `t.IsRawMode()`, `t.SetRawMode()` and `t.UnsetRawMode()`

Reply:
RawMode works really differently on windows/unix, so we'd have to bifurcate `readline.go` into a bunch of parts to handle that.